cpp/oneapi/dal/backend/primitives/objective_function/logloss_dpc.cpp

```diff
@@ -562,8 +570,17 @@ sycl::event logloss_hessian_product<Float>::compute_without_fit_intercept(
         element_wise(q_, kernel_mul, buf_ndview, hess_ndview, buf_ndview, { event_xv });
 
     auto event_xtdxv =
-        gemv(q_, data_.t(), buffer_, out, Float(1), L2_ * 2, { event_dxv, fill_out_event });
-    return event_xtdxv;
+        gemv(q_, data_.t(), buffer_, out, Float(1), Float(0), { event_dxv, fill_out_event });
+
+    const Float regul_factor = Float(L2_ * 2);
+
+    const auto kernel_regul = [=](const Float a, const Float param) {
```

Comment:
Why there is a duplication of this kernel?

Reply:
it is used in functions compute_without_fit_intercept and compute_withou_fit_intercept and from my point of view it is better to define these kernels locally rather than globally